src/parse.rs

```diff
@@ -137,17 +143,15 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
     }
 
     let fingerprint_collection = collection_guard.as_mut().unwrap();
+    let fingerprints = fingerprint_collection.get_mut_for_flavor(flavor);
 
     // Each addon dir mapped to fingerprint struct.
-    let new_fingerprints: FingerprintCollection = all_dirs
+    let new_fingerprints: Vec<_> = all_dirs
         .par_iter() // Easy parallelization
         .map(|dir_name| {
             let addon_dir = root_dir.join(dir_name);
             // If we have a stored fingerprint on disk, we use that.
-            if let Some(fingerprint) = fingerprint_collection
-                .iter()
-                .find(|f| &f.flavor == flavor && &f.title == dir_name)
-            {
+            if let Some(fingerprint) = fingerprints.iter().find(|f| &f.title == dir_name) {
```

Comment:
We no longer need to filter on flavor since this data set is now specific for the flavor